src/src/com/microsoft/aad/adal/HttpWebRequest.java

```diff
@@ -198,40 +201,6 @@ private static String convertStreamToString(InputStream is) throws IOException {
         }
     }
 
-    private static int getStatusCode(HttpURLConnection connection) throws IOException {
-        int statusCode = HttpURLConnection.HTTP_BAD_REQUEST;
-
-        try {
-            statusCode = connection.getResponseCode();
-        } catch (IOException ex) {
-
-            if (Build.VERSION.SDK_INT < 16) {
```

Comment:
It's actually in description to this PR.
We don't need to getResponseCode() twice anymore, since we call this method only after getInputStream90